### PR TITLE
1098460 default cmd

### PR DESCRIPTION
**Pull request by mihneadb**
Fixes bug 1098460.

Depends on #155. Those parts of the diff will disappear when #155 is merged.

TODO after merge:
- [ ] publish new schema

r? @gregarndt @jonasfj

### CODE REVIEW

**Comment by gregarndt** [comment]
This, and the other PR are looking good to me, I think we could probably be safe merging this.  Just asking @jonasfj for another set of eyes on the schema validation pieces.  I appreciate the help on this!

**Comment by mihneadb** [comment]
Sweet, no problem!

File: schemas/payload.js [hunk 1]
```diff
@@ -118,10 +118,10 @@ module.exports = {
     "command": {
       "title": "Docker command to run (see docker api).",
       "type": "array",
-      "minItems": 1,
       "items": {
         "type": "string"
       },
+      "default": [],
```
Comment (jonasfj) [inline schemas/payload.js]: Yeah, this looks right to me :)

**Comment by jonasfj** [comment]
problems. in PR #155 should be addressed first... Mainly use a global validator instance.

**Comment by mihneadb** [comment]
@jonasfj do you still see the purpose of the separate module that I was planning on unit testing? Since we only use one validator and that is from an external library, we shouldn't test it in this repo.
We should probably still test that the schema does what we expect it to do, so a minimal set of unit tests sound useful to me. What do you think?

**Comment by jonasfj** [comment]
> We should probably still test that the schema does what we expect it to do, so a minimal set of unit tests sound useful to me. What do you think?

Testing that the schema does what we expect is nice...
An added bonus is that if we change schema in the future, the errors from these tests will let us know if we broke backward compatibility.

So yeah, a minimal task set is perhaps sane. Since initialize `base.validator` is still so simple we can perhaps just duplicate that into the test case. I'm not entirely sure how to do it right, but I do something similar in other places were I test schemas.
`base.testing.schemas` can load schemas from a folder and test against a set of cases. But it won't work here because the schema is a javascript file that generates it, when loaded... So we have to use `require`.

**Comment by mihneadb** [comment]
@gregarndt @jonasfj I've updated the PR as discussed. Please review.

**Comment by jonasfj** [comment]
looks good to me...
@gregarndt, I hope you're into clicking the merge button :)
Assuming you have no objections.

**Comment by mihneadb** [comment]
Also don't forget to publish the new schema.

**Comment by gregarndt** [comment]
I'll try to get some time this weekend to check out the errors in the task graph, but it seems that tests are having a hard time now.

**Comment by mihneadb** [comment]
@gregarndt you're saying there is some unrelated aspect that made the tests fail?

**Comment by gregarndt** [comment]
It looks like the change to export the payload schema in task.js broke the import in task_listener.js.  I think this should fix it up:

https://github.com/gregarndt/docker-worker/commit/e58795adad699cf941297dc475b2d1916cf0362a

**Comment by mihneadb** [comment]
Oh wow :(. Should I cherry-pick that commit and try again?

**Comment by gregarndt** [comment]
Whatever works for you :).  However you add it, also make sure to rebase against master in case things have changed since then.

**Comment by mihneadb** [comment]
@gregarndt done! Thanks!

**Comment by gregarndt** [comment]
This looks much better now, thanks!

**Pull request merged**